### ISSUE TITLE
Linux version is poor

### ISSUE THREAD

**Issue by csimpi** [alexkay/spek#265]
Hi, 
thank you for this awesome tool.

Unfortunately, the installation readme doesn't tell what dependencies I need to install in order to compile the package on Linux. You should release binaries for Linux, I think you shouldn't expect all Linux users to pull up all the development packages just for this package.

**Comment by madzohan** [comment]
OFFTOP: if you want binary app to view spectrogram, there are a lot of alternatives, for example go get another tool named [tenacity](https://codeberg.org/tenacityteam/tenacity)  from [here](https://codeberg.org/attachments/cafc22ad-cb38-403e-ba6b-a38841b87b0e)

```
chmod +x ./tenacity-linux-v1.3-beta2-x86_64.appimage
sudo apt-get install libasound2 libasound2-dev
sudo ln -s /lib/x86_64-linux-gnu/libportaudio.so.2 /lib/x86_64-linux-gnu/libportaudio.so
./tenacity-linux-v1.3-beta2-x86_64.appimage
```

**Comment by csimpi** [comment]
@madzohan Yes this is the documentation that doesn't contain the information I am talking about, thx for cooperating with sharing the link to the documentation I was talking about ;)

**Comment by jlu5** [comment]
https://github.com/alexkay/spek/blob/master/INSTALL.md#building-from-the-tarball

> To build you will need wxWidgets and FFmpeg packages.

Telling us what distro you're using would be a good start. There's no one package name or command to install system libraries that covers all of Linux.

**Comment by csimpi** [comment]
@jlu5 seems you are not familiar with linux systems, no worries. So in order to build you need to install many-many development packages, really a lot. Development packages are not available by default and expecting normal users to install all of those just to use the app is unreasonable. 

"There's no one package name or command to install system libraries that cover all of Linux."
Ture, that's why I told, offering binaries would be the way to release your app versions.

**Comment by jlu5** [comment]
> @jlu5 seems you are not familiar with linux systems, no worries. So in order to build you need to install many-many development packages, really a lot. Development packages are not available by default and expecting normal users to install all of those just to use the app is unreasonable.

:roll_eyes: I've worked with Linux systems for almost a decade now. First time I've heard that accusation.

Unfortunately this means we sometimes forget to mention the implied build dependencies :) automake, g++ and friends. But you probably knew that already?

> 
> FYI: `wxWidgets and FFmpeg packages` are not the development packages I am talking about. Those are dependencies that OK to install even for average users, but installing these won't help with compiling.
> 
> "There's no one package name or command to install system libraries that cover all of Linux." Ture, that's why I told, offering binaries would be the way to release your app versions.

#256 is probably the best bet. The binary packages for spek have kind of rotted in recent years as activity moved between a couple of forks and then back to this repo. It got reintroduced in Debian for instance very recently (ie this month) after 6 years of inactivity!

**Comment by madzohan** [comment]
"Binary" like appimage also refers to system libs 😄 even self-contained flatpak apps could stop working after updating video card drivers 😑 (hello VLC-player 💩)

For example [tenacityaudio](https://tenacityaudio.org/) (and you could try it as a alternative to get specs):
```
$ uname -a
Linux kexix 5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy

$ chmod +x ./tenacity-linux-v1.3-beta2-x86_64.appimage
./tenacity-linux-v1.3-beta2-x86_64.appimage
```
in output it says - cannot find libportaudio.so lib ... but I have the installed libasound2 ... so the solution was 😄
```
$ sudo ln -s /lib/x86_64-linux-gnu/libportaudio.so.2 /lib/x86_64-linux-gnu/libportaudio.so
```

Anyway if you cant run something for some reason you could always get & run many alternatives which works for you 😜

**Comment by csimpi** [comment]
nice explanation of not being able to solve something that many other projects handle with no issue :) even very cross-platform libraries. this is the kinda first time in my life that somebody is trying to explain to be it's impossible to release a package to Linux without manually installing dependencies LOL. Also there's a thing called docker, you could offer a docker instance too :) there are many options to handle this... I mean you're really trying to tell me compiling from source is the ONLY way to release an app to Linux systems?

**Comment by madzohan** [comment]
I'm neither developer nor maintainer of this project, same user of opensource community world, but not the toxic one like you - who blame everyone of "not being able to solve something", when even not be able to provide normal details about his problem :man_facepalming: ... if you know about Docker - you should know about such things like: 1) your system info; 2) logs.

anyway if I were maintainer of this project I would simply forget about your problem and about your rotten personality, wish you good luck further in you meaningless life :kissing_heart:

**Comment by csimpi** [comment]
@madzohan this is my honest user review, nothing else. 
I think bumping here only just to insult me and give zero to the convo is very unprofessional and might be worst than my honest but not too positive opinion.

I still think the concern I raised here is legit, and the answers I got so far weren't satisfying, and still looking for answers/solutions. 

I don't understand your comment regarding using docker. I am happy to hear if you elaborate it more :)

**Comment by alexkay** [comment]
Flatpak packages are coming soon, someone else is working on it, see https://github.com/flathub/flathub/pull/3800

Many Linux distros already have packages in their main or auxiliary repositories. Some package managers allow installing build dependencies if you need them to build a newer/modified version from source, see for example `apt build-deps` for Debian derivatives.

The link mentioned above has build instructions: https://github.com/alexkay/spek/blob/master/INSTALL.md#building-from-the-tarball

I'm going to close this, I suggest you find another place to ask general Linux questions, for example: https://unix.stackexchange.com/

**Comment by csimpi** [comment]
@alexkay thank you, and sorry for the intermezzo

**Comment by illtellyoulater** [comment]
@csimpi is not wrong in stating that build instructions are incomplete. 

For example in my case, other than having to find out wxWidgets and FFmpeg specific package names for my distro and installing them, I also had to go through the error logs and understand myself I also had to install `autoconf`, `autopoint`, and `libtool-bin`, which are all not included in the provided building instructions at https://github.com/alexkay/spek/blob/master/INSTALL.md#building-from-the-tarball

...only to find out that I still couldn't compile the code, as the required version of `libavformat` library is too high and not yet supported in my distro (which has been going under a heavy reworking and is going to be rebased on the last Ubuntu version not sooner than a couple of months).

A Flatpak or AppImage would have solved all of this, yet no one seems to care here.

The lack of those alternatives, coupled with the lack for a complete guide with comprehensive, distro-specific compiling instructions, is the exact recipe to driving new linux users away... Sad to see how in 2024 there are still some of those more seasoned linux users/devs who are failing to realize this aspect, or just don't care about how detrimental this is to the whole linux world.
